### DISCUSSION TITLE
Using wildcard with watch

### DISCUSSION THREAD

**Discussion by FilipBrannlund** [react-hook-form/react-hook-form#5403] [Q&A] [answered]
I'm having an application where if the user checks the checkbox "I'm not satisfied" an watch is watching and scanning if other checkboxes below has been marked. If the user has not marked any checkboxes below an error message pops up saying "You need to pick atleast ONE reason below".
Everything works to this point.

Now when the message "You need to pick atleast ONE reason below" displays and the user marks any checkboxes below a new scan should execute and the error message should disapear.

I guess I could set up a watch in the Reason-component watching the 3 checkboxes but that component is getting iterated 1-500 times and that would mean 3 * 500 watchers which could have an impact on the perfomance i guess?

Is it possible in the Unsatisfied-component to define watch with wildcards?
Something like this
```
function Unsatisfied() {
  const { watch, getValues, control } = useFormContext(); // retrieve all hook methods
  const watchUnsatisfied = watch("unsatisfied", false);
  const watchColor = watch("car.*.color", false);
  const watchEngine = watch("car.*.engine", false);
  const watchWheels = watch("car.*.wheels", false);
```

Full example below:
https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-dj7x5?file=/src/App.js

If anyone has sees this could be coded in another more efficient way and have spare time over, please provide me with an CSB with improvements. I'm happy to learn. :)

**Comment by bluebill1049** [comment]
why not just `watch` the entire form? `watch()`

**Reply by FilipBrannlund** [reply]
Thanks! I did not think of watching the entire form.

**Comment by thanh-nguyen-95** [comment, marked as answer]
A complete solution with 1000 rows (fake data of course). Performance gets worse after 10K+

> https://codesandbox.io/s/custom-hook-with-resolver-v7-forked-84jwy?file=/src/App.js

What I did:

- Rename variables for clarity --> better readability
- Bring the _Send_ button to the top --> easier testing
- Remove `FormProvider` and pass anything needed through props --> easier to memoize component with `React.memo`
- Use `useFieldArray` & `useWatch` --> personal preferences
- Use native `<input>` without wrapping it in `Controller` --> reduce unnecessary code
- Use `React.memo` to memoize the `CarSection` --> optimze rendering cycle
- Make the `hasMarkedAnyReasons` check cleaner

The root cause is that components are re-rendered each time you check an option (doesn't matter which one).

**Reply by bluebill1049** [reply]
Awesome! 🥳

**Reply by bluebill1049** [reply]
for anything greater than 10k, perhaps @FilipBrannlund should consider https://react-hook-form.com/advanced-usage#Workingwithvirtualizedlists

**Reply by FilipBrannlund** [reply]
@thanh-nguyen-95 wow! Amazing thank you for an example, this will really help me so much!
